### ISSUE TITLE
Modern tree-sitter - watch configs

### ISSUE THREAD

**Issue by mauricioszabo** [pulsar-edit/pulsar#549]
### Have you checked for existing feature requests?

- [X] Completed

### Summary

In modern tree-sitter, on the highlights queries and other stuff we can make highlighting conditional by using `(#set! test.onlyIfConfig ...)`. But we don't _watch_ the config, so every editor that already highlighted code basically needs to re-highlight

We might want to watch the config so if we change the config, the highlighting works as expected

### What benefits does this feature provide?

We don't need to force a new highlighting on the editor when we change a config

### Any alternatives?

Select all, cut, then paste, but that's not a good experience

**Comment by savetheclocktower** [comment]
Yeah, this is a good idea. I think we could do it on an as-needed basis whenever the highlighter encounters a query with an `onlyIfConfig` predicate. We could even be smart enough to invalidate only those nodes when the given query changes, but that's probably overkill; easier to just invalidate the whole document.

We'd just want to make sure that we were listening only for the configured keys, rather than invalidation on any config change whatsoever.
